java/arcs/sdk/android/storage/service/StorageService.kt

```diff
@@ -71,6 +77,19 @@ open class StorageService : ResurrectorService() {
     private val stats = BindingContextStatsImpl()
     private val log = TaggedLog { "StorageService" }
 
+    var devToolsService: IDevToolsService? = null
+    val connection = object : ServiceConnection {
+        override fun onServiceConnected(className: ComponentName, service: IBinder) {
+            if (0 != applicationInfo.flags and ApplicationInfo.FLAG_DEBUGGABLE) {
+                devToolsService = IDevToolsService.Stub.asInterface(service)
+            }
+        }
+
+        override fun onServiceDisconnected(arg0: ComponentName) {
+            devToolsService = null
+        }
+    }
+
```

Comment:
This should probably not live here. I think it makes it too easy to accidentally hook in and include a DevToolsService when it shouldn't be included.

Instead - if StorageService.kt were to use an abstract factory - you could supply the implementation of that factory which creates the wrapping DevToolsBindingContext, whereas the normal implementation would supply the raw BindingContext.